core/src/test/scala/com/raphtory/BaseRaphtoryAlgoTest.scala

```diff
@@ -73,18 +76,33 @@ abstract class BaseRaphtoryAlgoTest[T: ClassTag: TypeTag]
       increment: Long,
       windows: List[Long]
   ): String = {
-    val startingTime         = System.currentTimeMillis()
     val queryProgressTracker =
       graph.rangeQuery(algorithm, outputFormat, start, end, increment, windows)
-    val jobId                = queryProgressTracker.getJobId()
+
+    val jobId = queryProgressTracker.getJobId()
+
     queryProgressTracker.waitForJob()
 
-    val dir     = new File(testDir + s"/$jobId").listFiles
+    val files = new File(outputDirectory + s"/$jobId")
+      .listFiles()
       .filter(_.isFile)
-    val results =
-      (for (i <- dir) yield scala.io.Source.fromFile(i).getLines().toList).flatten.sorted.flatten
-    val hash    = Hashing.sha256().hashString(new String(results), StandardCharsets.UTF_8).toString
+
+    val results = files.flatMap { file =>
+      val source = scala.io.Source.fromFile(file)
+      try source.getLines().toList
+      catch {
+        case e: Exception => throw e
+      }
+      finally source.close()
+    }.sorted
+
+    val hash = Hashing
+      .sha256()
+      .hashString(results.mkString, StandardCharsets.UTF_8)
+      .toString
+
     logger.info(s"Generated hash code: '$hash'.")
+
```

Comment:
factor this out into a separate function as it is used again below for `algorithmPointTest`